debugger/src/stree.c

```diff
@@ -464,7 +480,14 @@ void stree_add(GList *frames)
  */
 void stree_clear(void)
 {
+	/* removing nodes from tree triggers the "cursor-changed" event;
+	 * in this context we don't need to process it,
+	 * so the signal handler should be disconnected before the clear
+	 * and reconnected properly afterwards
+	 */
+	disable_cursor_changed_handler();
 	gtk_tree_store_clear(store);
+	enable_cursor_changed_handler();
```

Comment:
You probably should use [`g_signal_handler_block()`](https://docs.gtk.org/gobject/?q=g_signal_handler)-[family of functions](https://docs.gtk.org/gobject/?q=g_signal_handler) instead of rolling your own.